crates/rattler_package_streaming/src/read.rs

```diff
@@ -86,3 +88,60 @@ pub fn extract_conda(reader: impl Read, destination: &Path) -> Result<ExtractRes
 
     Ok(ExtractResult { sha256, md5 })
 }
+
+/// Extracts a folder from a tar.bz2 archive.
+pub fn folder_from_tar_bz2(
```

Comment:
Please put these in the `fs.rs` file because you directly access the filesystem.

Also, can you call them `extract_*` and rename `folder` to `directory`?